UATs Job with ambient integration Falied despite all notebook tests passed

### Bug Description

When the running the UATs with ambient integration, the UATs job reports status `Failed` even though the logs show that all the notebook tests passed. I bumped into this in the CI while working on https://github.com/canonical/charmed-kubeflow-solutions/pull/125. [Example run](https://github.com/canonical/charmed-kubeflow-solutions/actions/runs/22348927479/job/64670688918) of the failure in the CI.

### To Reproduce

Re-run the CI from the PR https://github.com/canonical/charmed-kubeflow-solutions/pull/125, or deploy ambient solution locally from the PR and run the `kubeflow-remote` UATs

### Environment

Microk8s 1.32/stable (v1.32.9)
Juju 3.6/stable (3.6.14)

### Relevant Log Output

```shell
INFO     test_kubeflow_workloads:test_kubeflow_workloads.py:352 Fetching Job logs...
INFO: detected pid 1, running init handler
{"logger":"","ts":"2026-02-24 11:56:55.061068","caller":{"file":"main.go","line":722},"level":0,"msg":"starting up","pid":11,"uid":65533,"gid":65533,"home":"/tmp","flags":["--add-user=false","--change-permissions=0","--cookie-file=false","--depth=1","--exechook-backoff=3s","--exechook-timeout=30s","--git=git","--git-gc=always","--group-write=true","--help=false","--http-metrics=false","--http-pprof=false","--link=charmed-kubeflow-uats","--man=false","--max-failures=0","--max-sync-failures=0","--one-time=true","--***","--period=10s","--ref=e833c86c98294fb9d58d549fdc200caacd19dca1","--repo=https://github.com/canonical/charmed-kubeflow-uats","--root=/tests","--ssh=false","--ssh-key-file=/etc/git-secret/ssh","--ssh-known-hosts=true","--ssh-known-hosts-file=/etc/git-secret/known_hosts","--stale-worktree-timeout=0s","--submodules=recursive","--sync-timeout=2m0s","--timeout=0","--v=-1","--verbose=0","--version=false","--wait=0","--webhook-backoff=3s","--webhook-method=POST","--webhook-success-status=200","--webhook-timeout=1s"]}
{"logger":"","ts":"2026-02-24 11:56:55.074967","caller":{"file":"main.go","line":1248},"level":0,"msg":"repo directory failed checks or was empty","path":"/tests"}
{"logger":"","ts":"2026-02-24 11:56:55.075125","caller":{"file":"main.go","line":1258},"level":0,"msg":"initializing repo directory","path":"/tests"}
{"logger":"","ts":"2026-02-24 11:56:56.698669","caller":{"file":"main.go","line":1797},"level":0,"msg":"update required","ref":"e833c86c98294fb9d58d549fdc200caacd19dca1","local":"","remote":"e833c86c98294fb9d58d549fdc200caacd19dca1","syncCount":0}
{"logger":"","ts":"2026-02-24 11:56:58.579478","caller":{"file":"main.go","line":1848},"level":0,"msg":"updated successfully","ref":"e833c86c98294fb9d58d549fdc200caacd19dca1","remote":"e833c86c98294fb9d58d549fdc200caacd19dca1","syncCount":1}
{"logger":"","ts":"2026-02-24 11:56:58.579640","caller":{"file":"main.go","line":1022},"level":0,"msg":"exiting after one sync","status":0}
Defaulted container "test-kubeflow" out of: test-kubeflow, git-sync (init)
============================= test session starts ==============================
platform linux -- Python 3.11.11, pytest-9.0.2, pluggy-1.5.0 -- /opt/conda/bin/python3.11
cachedir: .pytest_cache
rootdir: /tests/.worktrees/e833c86c98294fb9d58d549fdc200caacd19dca1/tests
configfile: pytest.ini
plugins: anyio-4.8.0
collecting ... collected 11 items / 5 deselected / 6 selected

test_notebooks.py::test_notebook[katib-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:55 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU, use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`. To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:68 Running katib-integration.ipynb...
PASSED                                                                   [ 16%]
test_notebooks.py::test_notebook[kfp-v2-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:55 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU, use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`. To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:68 Running kfp-v2-integration.ipynb...
PASSED                                                                   [ 33%]
test_notebooks.py::test_notebook[kserve-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:55 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU, use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`. To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:68 Running kserve-integration.ipynb...
PASSED                                                                   [ 50%]
test_notebooks.py::test_notebook[test-pod-security-standards] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:55 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU, use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`. To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:68 Running test-pod-security-standards.ipynb...
PASSED                                                                   [ 66%]
test_notebooks.py::test_notebook[training-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:55 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU, use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`. To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:68 Running training-integration.ipynb...
PASSED                                                                   [ 83%]
test_notebooks.py::test_notebook[kubeflow-trainer-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:55 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU, use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`. To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:68 Running kubeflow-trainer-integration.ipynb...
PASSED                                                                   [100%]

================= 6 passed, 5 deselected in 2601.46s (0:43:21) =================
##### git-sync initContainer logs #####
##### test-kubeflow container logs #####
FAILED

=================================== FAILURES ===================================
___________________________ test_kubeflow_workloads ____________________________
Traceback (most recent call last):
  File "/home/ubuntu/charmed-kubeflow-uats/driver/test_kubeflow_workloads.py", line 345, in test_kubeflow_workloads
    wait_for_job(lightkube_client, JOB_NAME, NAMESPACE)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/tenacity/__init__.py", line 338, in wrapped_f
    return copy(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/tenacity/__init__.py", line 477, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/tenacity/__init__.py", line 378, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/tenacity/__init__.py", line 400, in <lambda>
    self._add_action_func(lambda rs: rs.outcome.result())
                                     ^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/actions-runner/_work/_tool/Python/3.12.12/x64/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/actions-runner/_work/_tool/Python/3.12.12/x64/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/tenacity/__init__.py", line 480, in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/driver/utils.py", line 110, in wait_for_job
    raise ValueError(f"Job {namespace}/{job_name} failed!")
ValueError: Job test-kubeflow/test-kubeflow failed!

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/runner.py", line 344, in from_call
    result: TResult | None = func()
                             ^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
    yield
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/capture.py", line 900, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/skipping.py", line 263, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/python.py", line 1670, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 53, in run_old_style_hookwrapper
    return result.get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_result.py", line 103, in get_result
    raise exc.with_traceback(tb)
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 38, in run_old_style_hookwrapper
    res = yield
          ^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/python.py", line 156, in pytest_pyfunc_call
    result = testfunction(**testargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/charmed-kubeflow-uats/driver/test_kubeflow_workloads.py", line 347, in test_kubeflow_workloads
    pytest.fail(
  File "/home/ubuntu/charmed-kubeflow-uats/.tox/kubeflow-remote/lib/python3.12/site-packages/_pytest/outcomes.py", line 177, in fail
    raise Failed(msg=reason, pytrace=pytrace)
Failed: Something went wrong while running Job test-kubeflow/test-kubeflow. Please inspect the attached logs for more info...
```

### Additional Context

_No response_

--- a/assets/test-job.yaml.j2
+++ b/assets/test-job.yaml.j2
@@ -6,6 +6,10 @@ spec:
   backoffLimit: 0
   template:
     metadata:
+      annotations:
+        # Ensures istio-proxy container terminates when the Job finishes (sidecar)
+        # and is safely ignored in ambient mode.
+        sidecar.istio.io/nativeSidecar: "true"
       labels:
           {% if proxy %}
           notebook-proxy: "true"
@@ -47,26 +51,7 @@ spec:
               cd /tests/charmed-kubeflow-uats/tests;
               {% endif %}
               python3 -m pip install -r requirements.txt >/dev/null;
-              {{ pytest_cmd }};
-              # Kill Istio Sidecar after workload completes to have the Job status properly updated
-              # https://github.com/istio/istio/issues/6324
-              x=$(echo $?);
-
-              # Kill Istio sidecar -- does the same function as the commented out curl command below
-              # curl -fsI -X POST http://localhost:15020/quitquitquit >/dev/null && exit $x;
-              # The curl command was replaced by the Python code below to support this step in images that don't have curl installed
-              python3 - <<'EOF'
-              import http.client
-
-              try:
-                  conn = http.client.HTTPConnection("localhost", 15020, timeout=2)
-                  conn.request("POST", "/quitquitquit")
-                  conn.close()
-              except Exception:
-                  pass
-              EOF
-
-              exit $x
+              {{ pytest_cmd }}
           volumeMounts:
             - name: test-volume
               mountPath: /tests

